### ISSUE TITLE
Attributes derived from IgnoreAttribute are ignored

### ISSUE THREAD

**Issue by gameleon-dev** [praeclarum/sqlite-net#558]
Since version 1.3.3 of the library there has been an issue with using attributes that derive from IgnoreAttribute.

The following example works fine:

```
[Table]
public class Person
{
      //......
      [Ignore]
      public string Name { get; set; }
}
```

The following example fails (Name is still added to the table):

```
[AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
public class MyIgnoreAttribute : IgnoreAttribute
{
           
}

[Table]
public class Person
{
      
     //....

      [MyIgnore]
      public string Name { get; set; }
}
```

Could this be a issue with the new reflection API used in the library?

**Comment by gameleon-dev** [comment]
Found the culprit of this issue.

**SQLite.cs, Line 1889:**
`var ignore = p.CustomAttributes.Any(x => x.AttributeType == typeof(IgnoreAttribute));`

This kind of check only checks for the `IgnoreAttribute` itself, but not for attributes derived from `IgnoreAttribute`

Changing this line to any of the following solves the issue

`var ignore = p.CustomAttributes.Any(x => x.AttributeType.IsAssignableFrom(typeof(IgnoreAttribute)));`

or 

`var ignore = p.IsDefined(typeof(IgnoreAttribute),true);`

The second option seems like the cleaner solution.

**Comment by gameleon-dev** [comment]
I have created a pull request to fix the issue over here: #559

**Comment by hvaughan3** [comment]
@praeclarum @gameleon-dev  
Possibly related to my [post here](https://github.com/praeclarum/sqlite-net/issues/549#issuecomment-315137240)?

**Comment by praeclarum** [comment]
Thanks for catching this. Working on a test now...

**Comment by gameleon-dev** [comment]
@praeclarum @hvaughan3 

That seems related to this issue, yes. I verified it in my code and it seems the already merged pull request #559 solves issues like #549 as well.

**Comment by praeclarum** [comment]
Thanks! I added a test so hopefully won't regress on this again.
